topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorTest.java

```diff
@@ -151,6 +151,22 @@ public void testOnKafkaTopicAdded_ignorable(VertxTestContext context) {
         context.completeNow();
     }
 
+    /** Test what happens when a KafkaTopic with no spec gets created in kubernetes */
+    @Test
+    public void testOnKafkaTopicAdded_nospec(VertxTestContext context) {
```

Comment:
```suggestion
    public void testOnKafkaTopicAddedNoSpec(VertxTestContext context) {
```

let's use camelcase ...

Reply:
Using underscore is also a common format in unit and/or bdd testing. Esp. it is very common in tests formatted as given-when-then (a quick example here: https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea).

I just followed the convention which is already used in the methods here.

I will reword the "nospec" to "noSpec" but wanna leave the _ as the other methods also have it, if you don't mind.

Reply:
When the method names include test-specific states (such as given-when-then, or simply before-after states), they might get longer so in order to make readability better it is common to use underscores along with the camel style in Java.

In languages like Python, since _ usage is common, they use two underscores (__) instead. (test_on_kafka_topic_added__no_spec). So the _ which is used in test methods to separate the behavioral situations is sth language independent. It is the space conversions that vary from language to language. Java use camel case, Python uses snake case, and some use kabab case (I guess it was Lisp lang IIRC).

Reply:
Tbh I have seen this in other languages (as you said in Python) but not with Java. Anyway I am not going to "block" this PR just for this :-) if the others are ok with the naming.

Reply:
Tbh I have seen this in other languages (as you said in Python) but not with Java. Anyway I am not going to "block" this PR just for this :-) if the others are ok with the naming.

Reply:
I wanted to make them consistent within the test class. As you might notice, the other methods are named just with camel case. 

Since both methods are not that long and doesnt include a huge scenario normally I would start with a normal convention (like the last one you mention here), and when the scenarios get more complex, I would start using _ for the step state separation.

I just wanted to emphasize that using underscore is sth common even in Java, but in the cases of using scenarios. Here we have a one-two step scenario which doesn't make much difference to use the generic convention or the behavioral-test special one.

If you don't mind, we can reach out to Tom later on and ask him about why he preferred to use that convention in one test class, but not the other one as both seem to include a state change (thus a +1 step scenario)

Reply:
BTW for the overall style thing, we can discuss and create a subset of team agreement document (I say subset because team agreement docs not only have techie bits but everything related to teams/groups) that would fit the contributor guide of this project so that anyone who contributes would follow the guide. I will try to propose it in one of the community meetings (I guess that would be the right place)

Reply:
I am not sure it's worth doing the effort on an agreement doc about code style. I was just asking why the difference. Knowing the reason from @tombentley would be enough for me :-)

Reply:
What I mean is a kinda social contract that includes the small technical bits, which might help resolve conflicts easier (not for just style-wise concerns of course). I believe it is worth doing it, as it is not just for code style but for general consistency.

With all respect, leaving a link here for further reading as anyone might be interested in the topic as well: https://openpracticelibrary.com/practice/social-contract/